### PR TITLE
add create conversation to CloudAdapter

### LINKED ISSUE #5488: Add CreateConversationAsync to CloudAdapter implementation
```cs
                await ((BotFrameworkAdapter)turnContext.Adapter).CreateConversationAsync(
                    teamsChannelId,
                    serviceUrl,
                    credentials,
                    conversationParameters,
                    async (t1, c1) =>
                    {
                        conversationReference = t1.Activity.GetConversationReference();
                        await ((BotFrameworkAdapter)turnContext.Adapter).ContinueConversationAsync(
                            _appId,
                            conversationReference,
                            async (t2, c2) =>
                            {
                                await t2.SendActivityAsync(proactiveMessage, c2);
                            },
                            cancellationToken);
                    },
                    cancellationToken);
```

https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/57.teams-conversation-bot/Bots/TeamsConversationBot.cs#L150

### PR DESCRIPTION
**Pull request by johnataylor**
Fixes #5488

Also updates the abstract class to avoid casts in the application code.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder/BotAdapter.cs [hunk 1]
```diff
@@ -242,6 +242,33 @@ public virtual Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, Act
...
+        /// and the user's account information on that channel.
+        /// Most _channels only support initiating a direct message (non-group) conversation.
+        /// <para>The adapter attempts to create a new conversation on the channel, and
+        /// then sends a <c>conversationUpdate</c> activity through its middleware pipeline
+        /// to the <paramref name="callback"/> method.</para>
+        /// <para>If the conversation is established with the
+        /// specified users, the ID of the activity's <see cref="IActivity.Conversation"/>
+        /// will contain the ID of the new conversation.</para>
+        /// </remarks>
+        public virtual Task CreateConversationAsync(string botAppId, string channelId, string serviceUrl, ConversationParameters conversationParameters, BotCallbackHandler callback, CancellationToken cancellationToken)
```
Comment (stevengum) [inline libraries/Microsoft.Bot.Builder/BotAdapter.cs:267]: Is there a specific reason we didn't have this method before? Or any of the CreateConversationAsync implementations in `BotFrameworkAdapter`?

Will this support Skills making 1:1 conversations on behalf of the host bot (e.g. in Teams)?

@johnataylor @EricDahlvang

Comment (EricDahlvang) [reply libraries/Microsoft.Bot.Builder/BotAdapter.cs:267]: @stevengum I'm not sure about Skills, but this should enable CloudAdapter in this Teams scenario: https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/57.teams-conversation-bot/Bots/TeamsConversationBot.cs#L150

**Comment by tracyboehrer** [comment]
Is this an R14 goal?  I only ask because at least for the Teams conversation sample, this would be required to get around my hack (which we could never release).  I can't say whether the Composer Teams samples are on the slate for R14, so it may not matter.

File: libraries/Microsoft.Bot.Builder/BotAdapter.cs (hunk 1 above)
Comment (johnataylor) [reply libraries/Microsoft.Bot.Builder/BotAdapter.cs:267]: I think it was just an omission. This improves the signature slightly. Its probably fair to say we simply don't have enough examples to make strong claims around how general any of this is. It improves the typing (removes the casts) by adding it to the abstract class and makes things more consistent. This works for Teams.

**Comment by johnataylor** [comment]
> Is this an R14 goal? I only ask because at least for the Teams conversation sample, this would be required to get around my hack (which we could never release). I can't say whether the Composer Teams samples are on the slate for R14, so it may not matter.

**Comment by johnataylor** [comment]
@tracyboehrer yes absolutely R14. Precisely because the gap you identified! Thanks!

**Review by EricDahlvang — APPROVED** [review]
This works for Teams.  However, will this work from a skill without also exposing 'audience'? (as pointed out by @stevengum )

**Pull request merged**